### ISSUE TITLE
[jsx-child-element-spacing] Improve error location

### ISSUE THREAD

**Issue by nwoltman** [yannickcr/eslint-plugin-react#1660]
With rule `react/jsx-child-element-spacing`, the error location is not very helpful when there is ambiguous spacing before the inline element.

**Current:**

```jsx
<div> {/* Error is at the end of this element */}
  Some
  text
  <a href="/">
    link
  </a>
</div>
```

**Expected:**

```jsx
<div>
  Some
  text
  <a href="/"> {/* Error should be at the start of this element */}
    link
  </a>
</div>
```

**Comment by ljharb** [comment]
cc @pfhayes

**Comment by pfhayes** [comment]
Addressed in https://github.com/yannickcr/eslint-plugin-react/pull/1666

**Comment by nwoltman** [comment]
Thanks @pfhayes!
